ipapython/certdb.py

```diff
@@ -156,10 +156,12 @@ def import_pkcs12(self, pkcs12_filename, db_password_filename,
                 "-i", pkcs12_filename,
                 "-k", db_password_filename, '-v']
         if pkcs12_passwd is not None:
-            pkcs12_passwd = pkcs12_passwd + '\n'
-            args = args + ["-w", paths.DEV_STDIN]
+            pkcs12_password_file = ipautil.write_tmp_file(pkcs12_passwd)
+            args = args + ["-w", pkcs12_password_file.name]
+        else:
+            pkcs12_password_file = None
```

Comment:
Nitpick shield activated: having it inside the else branch saves you one assignment operation. But I don't really care, so let's have it your way.